src/loaders/xml/mod.rs

```diff
@@ -0,0 +1,39 @@
+mod client_info;
+
+use serde::Deserialize;
+use serde_xml_rs::de::Deserializer;
+use xml::reader::{EventReader, ParserConfig};
+
+pub use self::client_info::{ClientInfo, Connection};
+use super::GameFileLoader;
+#[cfg(feature = "debug")]
+use crate::debug::*;
+
+#[derive(Default)]
+pub struct XMLFileLoader;
+
+impl XMLFileLoader {
+    pub fn load_client_info(&self, game_file_loader: &mut GameFileLoader) -> ClientInfo {
+        let clientinfo = game_file_loader.get("data\\clientinfo.xml");
+        let sclientinfo = game_file_loader.get("data\\sclientinfo.xml");
+
+        let src = match (sclientinfo, clientinfo) {
+            (Ok(data), _) => String::from_utf8(data).unwrap(),
+            (_, Ok(data)) => String::from_utf8(data).unwrap(),
+            _ => panic!("FOO"),
+        };
```

Comment:
I remember @vE5li mentioned to read one before reading another one (if it doesn't exist). In this case, what would you suggest to?

Reply:
This can be very nicely expressed with some functional style Rust
```rust
let clientinfo = game_file_loader
    .get("data\\sclientinfo.xml")
    .or_else(|_| game_file_loader.get("data\\clientinfo.xml"))
    .expect("failed to find clientinfo");
```
In this case, we only load `clientinfo.xml` if `sclsclientinfo.xml` was not found, thereby saving some computation. And since all of this works on `Result`s we can easily `expect` that at least one of them is present.